src/coreclr/gc/gcpriv.h

```diff
@@ -5376,13 +5374,13 @@ class region_allocator
 
 public:
     bool init (uint8_t* start, uint8_t* end, size_t alignment, uint8_t** lowest, uint8_t** highest);
-    bool allocate_region (size_t size, uint8_t** start, uint8_t** end);
+    bool allocate_region (size_t size, uint8_t** start, uint8_t** end, int direction);
     bool allocate_basic_region (uint8_t** start, uint8_t** end);
-    bool allocate_large_region (uint8_t** start, uint8_t** end);
+    bool allocate_large_region (uint8_t** start, uint8_t** end, int direction);
```

Comment:
instead of a value 1 or -1, since this is an interface to the rest of the GC code I would suggest to make an enum -

```
enum allocate_direction
{
    allocate_forward = 1,
    allocate_backward = -1,
};
```